examples/02-plot/sphere_eversion.py

```diff
@@ -0,0 +1,263 @@
+    Output h goes from -infinity to infinity, phi goes from -phi to phi.
+
+    """
+    h = w * np.sin(theta) / np.cos(theta) ** n
+    # phi is unchanged
+    return h, phi
+
+
+def cylinder_to_wormhole(h, phi, t, p, q):
+    """Map from a cylinder to an open wormhole using Eq. (4).
+
+    Input h goes from -infinity to infinity, phi goes from -phi to phi.
+    Output is an (x, y, z) point embedded in 3d space.
+
+    The parameters t, p, q vary during the eversion process.
+    Start from |t| > 1 (fixed), p = 1 and q = 0. End at p = 0, qt = +-1.
+
+    """
+    x = t * np.cos(phi) + p * np.sin((n - 1) * phi) - h * np.sin(phi)
+    y = t * np.sin(phi) + p * np.cos((n - 1) * phi) + h * np.cos(phi)
+    z = h * np.sin(n * phi) - t / n * np.cos(n * phi) - q * t * h
+    return x, y, z
+
+
+def close_wormhole(x0, y0, z0, eta, xi, alpha):
+    """Close the wormhole using Eqs. (7)-(8).
+
+    Input is an (x0, y0, z0) point embedded in 3d space.
+    Output is an (x2, y2, z2) == (x'', y'', z'') point embedded in 3d space.
+
+    The parameters eta, xi, alpha vary during the eversion process.
+
+    """
+    # Eq. (7): (x, y, z) -> (x', y', z')
+    denominator = xi + eta * (x**2 + y**2)
+    x1 = x0 / (denominator**kappa)
+    y1 = y0 / (denominator**kappa)
+    z1 = z0 / denominator
+
+    gamma = 2 * np.sqrt(alpha * beta)
+    # singular case, Eq (9):
+    if np.isclose(gamma, 0):
+        denominator = x1**2 + y1**2
+        x2 = x1 / denominator
+        y2 = y1 / denominator
+        z2 = -z1
+        return x2, y2, z2
+
+    # Eq. (8): (x', y', z') -> (x'', y'', z'')
+    exponential = np.exp(gamma * z1)
+    numerator = alpha - beta * (x1**2 + y1**2)
+    denominator = alpha + beta * (x1**2 + y1**2)
+    x2 = x1 * exponential / denominator
+    y2 = y1 * exponential / denominator
+    z2 = numerator / denominator * exponential / gamma - (alpha - beta) / (alpha + beta) / gamma
+    return x2, y2, z2
+
+
+def unfold_sphere(theta, phi, t, q, eta, lamda):
+    """Unfold the sphere using Eqs. (12), (15), (10).
+
+    Input is a (theta, phi) point in spherical coordinates.
+    Output is an (x, y, z) point embedded in 3d space.
+
+    The parameter lamda varies. Lamda = 1 is the final stage of the
+    wormhole closing, and lamda = 0 is the recovered sphere.
+
+    """
+    # apply Eqs. (12), (15)
+    # fmt: off
+    x = (
+        t * (1 - lamda + lamda * np.cos(theta)**n) * np.cos(phi)
+        - lamda * w * np.sin(theta) * np.sin(phi)
+    )
+    x /= np.cos(theta)**n
+    y = (
+        t * (1 - lamda + lamda * np.cos(theta)**n) * np.sin(phi)
+        + lamda * w * np.sin(theta) * np.cos(phi)
+    )
+    y /= np.cos(theta) ** n
+    z = (
+        lamda * (
+            (w * np.sin(theta) * (np.sin(n * phi) - q * t)) / np.cos(theta)**n
+            - t / n * np.cos(n * phi)
+        )
+        - (1 - lamda) * eta**(1 + kappa) * t * abs(t)**(2 * kappa)
+            * np.sin(theta) / np.cos(theta)**(2 * n)  # noqa
+    )
+    # fmt: on
```

Comment:
Disabling linting is never great form, but complex math equations like these are the epitomy of the kind of code that `black` butchers. Here are the equations:
![equation (12) from the paper](https://user-images.githubusercontent.com/17914410/183384613-de873a49-db39-4f88-ae6b-b7828fd5eec6.png)
![equation (15) from the paper](https://user-images.githubusercontent.com/17914410/183384635-1304da87-3037-47fb-9079-bb96e67b455b.png)

And the `# noqa` directive is necessary in order to have additional indentation on the corresponding line, otherwise flake8 complains.

Reply:
Agreed. Black is good for 99%, but these long multi-line equations look terrible when less than 100 char.

Reply:
@adeak You maybe know it but GitHub is now supporting [TeX rendering](https://github.blog/2022-05-19-math-support-in-markdown/).

$$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$

Reply:
Thanks @tkoyama010, I did know that. But it would've been tedious to retype those equations myself (and even if the arxiv version has the tex source, it might use custom commands making copy-pasting also tedious), and it would've opened the possibility for typos in my transcription, which could've led to false positive discussions in review here because the code might not match my (wrong) transcription.